dashboard/app/controllers/api/v1/pd/regional_partner_workshops_controller.rb

```diff
@@ -18,26 +18,25 @@ def find
 
     # Find the matching partner, even if it has no workshops
     partner = @partners.find_by_region(zip_code, state) || RegionalPartner.find_by_region(zip_code, state)
-    render json: serialize_partner_workshops(partner)
+    if partner.nil?
+      # TODO: should we instead render json which has the keys but nil values, to preserve previous behavior?
```

Comment:
There are unit test cases (e.g. `find converts full state name into 2 letter code` in `RegionalPartnerWorkshopsControllerTest`) which hit the case where `partner` is nil. Those tests don't assert on the response, so I wasn't sure if needed to preserve the previous behavior.

Reply:
haha in the comments `TODO (Andrew): Look into updating our very outdated version of ActiveModelSerializers`

Reply:
Does this fail with the new version when partner is nil? I can't 100% tell if we'd have any issues with no keys on the JS side.
```
render json: partner,
             serializer: Api::V1::Pd::RegionalPartnerWorkshopsSerializer,
             scope: {course: @course, subject: @subject}
```

Reply:
It does, unfortunately:

```
ERROR["test_find_preserves_2_letter_state_codes", "Api::V1::Pd::RegionalPartnerWorkshopsControllerTest", 14.87590100010857]
 test_find_preserves_2_letter_state_codes#Api::V1::Pd::RegionalPartnerWorkshopsControllerTest (14.88s)
NoMethodError:         NoMethodError: undefined method `model_name' for NilClass:Class
            app/controllers/api/v1/pd/regional_partner_workshops_controller.rb:25:in `find'
            app/controllers/application_controller.rb:167:in `block in with_locale'
            app/controllers/application_controller.rb:166:in `with_locale'
            app/helpers/read_replica_helper.rb:20:in `set_read_only_connection_for_block'
            test/controllers/api/v1/pd/regional_partner_workshops_controller_test.rb:145:in `block in <class:RegionalPartnerWorkshopsControllerTest>'
            test/testing/setup_all_and_teardown_all.rb:22:in `run'
```

Reply:
> I can't 100% tell if we'd have any issues with no keys on the JS side.

@bencodeorg @molly-moen 

Do you know if we can test this case manually? Or whether this case even comes up?

Reply:
I'm pretty sure that's a case we'd hit, let me investigate a bit...

Reply:
I don't know exactly how to test this manually right now, as applications are currently closed (so I can't really see it used in production) and I'm not 100% sure how to set up applications to be open locally. Applications are a big blind spot for Molly and me currently, as the system is generally most actively developed in the fall, and we both weren't on the team last fall. I think she commented separately that if this breaks anything, we'll find out about it over the next couple months during active development on applications.

Reply:
Since not every application can be matched to a partner, I think this is a case we would hit. I can see about enabling applications locally to test this out tomorrow.

Reply:
@clareconstantine @bencodeorg Let me just change it to preserve existing behavior, then. Added a unit test in a separate PR: https://github.com/code-dot-org/code-dot-org/pull/36646

If we can merge that first, then I can merge from staging here and change the behavior to make sure the test still passes.

Reply:
Updated this here https://github.com/code-dot-org/code-dot-org/pull/36560/commits/bb374051520d1f108ccb98a89d917f7d244058b8